code/workspaces/web-app/src/hooks/assetRepoHooks.js

```diff
@@ -2,12 +2,27 @@ import useShallowSelector from './useShallowSelector';
 import assetRepoSelectors from '../selectors/assetRepoSelectors';
 import sortByName from '../components/common/sortByName';
 
-export const useAssetRepo = () => useShallowSelector(assetRepoSelectors.assetRepo);
+const useAssetRepoUnsorted = () => useShallowSelector(assetRepoSelectors.assetRepo);
+
+export const useAssetRepo = () => {
+  const assetRepo = useAssetRepoUnsorted();
+  return {
+    ...assetRepo,
+    value:
+    {
+      assets: sortByName(assetRepo.value.assets),
+    },
+  };
+};
+
+const assetVisible = (projectKey, asset) => asset.fileLocation
+  && (asset.visible === 'PUBLIC'
+  || (asset.projects && asset.projects.filter(project => project.key === projectKey).length > 0));
```

Comment:
this might be clearer as 

```js
  || (asset.projects && asset.projects.includes(projectKey)));
```

Reply:
But projects isn't an array of strings, it's an array of structures.  I'll change it to
```javascript
|| (asset.projects && asset.projects.map(project=>project.key).includes(projectKey)
```

Reply:
Oh of course it is, that was silly of me sorry! Okay, I'm happy with that change